### PR TITLE
Fix assigning (somewhat) add basic popup

### LINKED ISSUE #61: Make assigning actually work
#58  also related

- should take up a 'slot' in the room
- resource acquisition rate should increase when worker is assigned
- crew should move to a specific tile within the room rather than wherever the user clicks

### LINKED ISSUE #62: Rework rooms to add Area2D, add room popup like we have with the crew popup
- make them clickable rather than clicking particular point on the screen related to the camera
- be able to detect when crew enter etc
- possibly remove crew collisions while in rooms etc
- cycle through current rooms
- show resource data etc

### PR DESCRIPTION
**Pull request by conalli**
- assigning now moves crew to hotspot
- add basic room popup
- could not get area2d to work correctly so reused the original method by calculating the selected tile
- general small refactoring, moved some things around

the bug with moving the camera still exists, will need further investigation

closes #61, closes #62

### CODE REVIEW

File: scripts/resource/RoomData.gd [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
+	room_type: RoomType = RoomType.new(), # Instantiate a blank RoomType
+	room_top_left: Vector2i = Vector2i(0, 0),
+	room_bottom_right: Vector2i = Vector2i(0, 0)
 ):
-	id = p_id
-	room_type = p_room_type
-	top_left = p_top_left
-	bottom_right = p_bottom_right
+	id = room_id
+	room_type = room_type
```
Comment (khatakslied) [inline scripts/resource/RoomData.gd:18]: Do we need self.room_type on the left if the names match?

File: scripts/resource/RoomData.gd [hunk 2]
```diff
@@ -35,16 +37,16 @@ func calculate_tile_count(vector1: Vector2i, vector2: Vector2i) -> int:
...
 	hot_spots.clear()
-	var capacity = room_type.capacity
-	var tile_count = calculate_tile_count(top_left, bottom_right)
+	var capacity = type.capacity
+	var _tile_count = calculate_tile_count(top_left, bottom_right)
 	var available_tiles = []
 	
 	var min_x = min(top_left.x, bottom_right.x)
-	var max_x = max(top_left.x, bottom_right.x) + 1
+	var _max_x = max(top_left.x, bottom_right.x) + 1
```
Comment (khatakslied) [inline scripts/resource/RoomData.gd:49]: Why only for this one?

File: scripts/resource/Station.gd [hunk 3]
```diff
@@ -89,5 +89,12 @@ func add_room(room: Room) -> void:
...
 	rooms.erase(room)
+	room.queue_free()
 	update_max_hydrogen()
 	update_power()
+
+func find_room_by_id(room_id: int) -> Room:
+	for room in rooms:
+		if room.data.id == room_id:
+			return room
+	return null
```
Comment (khatakslied) [inline scripts/resource/Station.gd:100]: Very useful. Thank you!

File: scripts/resource/RoomData.gd (hunk 1 above)
Comment (conalli) [reply scripts/resource/RoomData.gd]: you're right i missed updating the name for this one

File: scripts/resource/RoomData.gd (hunk 2 above)
Comment (conalli) [reply scripts/resource/RoomData.gd:49]:
```suggestion
        var _tile_count = calculate_tile_count(top_left, bottom_right)
        ...
	var _max_x = max(top_left.x, bottom_right.x) + 1
```
not just for that one, there were two variable that were not being used but i wasn't sure if they should be removed or if you planned on using them for something later on so i left it. the underscore marks it as an unused variable and stops warnings being shown.

Comment (khatakslied) [reply scripts/resource/RoomData.gd:49]: @conalli Sorry. I didn't fix this line after trying to figure out why the range wasn't working:

`for x in range(min_x, bottom_right.x + 1):`

It should be max_x here.

**Review by khatakslied — APPROVED** [review]
Looks great! I'll leave it for @jazmand to take a look :)

**Pull request merged**